app/src/main/java/de/danoeh/antennapod/adapter/DownloadLogAdapter.java

```diff
@@ -165,4 +174,16 @@ public long getItemId(int position) {
         DownloadStatus getItem(int position);
     }
 
+    public void retryAll() {
+        for (int i = 0; i < getCount(); i++) {
+            DownloadStatus status = getItem(i);
+            if (status != null && status.getFeedfileType() == FeedMedia.FEEDFILETYPE_FEEDMEDIA) {
+                retryDownloadSingleItem(status.getFeedfileId(), false);
+            }
+        }
+        ((MainActivity) context).showSnackbarAbovePlayer(
+                R.string.status_downloading_multiple_label, Toast.LENGTH_SHORT);
+        notifyDataSetChanged();
```

Comment:
Why is this needed? I think the list is not changed at this point. Have I overlooked something?